### PR TITLE
feat(dockfmt): add comment support

### PR DESCRIPTION
**Pull request by hyyking**
keeping comments and adding a line jump before it

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# This file lists all individuals having contributed content to the repository.
```
Comment (ferhatelmas) [inline AUTHORS]: is this really needed? contributor graph is there

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,8 @@
 FROM golang:alpine as builder
 MAINTAINER Jessica Frazelle <jess@linux.com>
 
+# Comment
```
Comment (ferhatelmas) [inline Dockerfile]: since it doesn't add value, I wouldn't add it.

File: format.go [hunk 3]
```diff
@@ -119,13 +119,11 @@ func (cmd *formatCommand) Run(ctx context.Context, args []string) error {
 func (df *file) doFmt(ast *parser.Node) (result string, err error) {
 	// check if we are on the correct line,
 	// otherwise get the comments we are missing
```
Comment (ferhatelmas) [inline format.go]: comment is also outdated

File: format.go [hunk 4]
```diff
@@ -161,18 +159,29 @@ func (df *file) doFmt(ast *parser.Node) (result string, err error) {
...
 	return
 }
 
+func (df *file) getCommentsUntil(until int) (result string, err error) {
+	comments, err := df.getOriginalLines(df.currentLine, until, df.name)
+	if err != nil {
+		return "", err
+	}
+
+	re := regexp.MustCompile(`#.*\n`)
```
Comment (ferhatelmas) [inline format.go]: this might be an issue for windows but enough for our purposes.

File: Dockerfile (hunk 2 above)
Comment (hyyking) [reply Dockerfile]: It was to test the script will remove it

File: format.go (hunk 4 above)
Comment (hyyking) [reply format.go]: how come ?

File: format.go (hunk 3 above)
Comment (hyyking) [reply format.go]: updating

File: AUTHORS (hunk 1 above)
Comment (hyyking) [reply AUTHORS]: generated by Makefile

File: format.go (hunk 4 above)
Comment (ferhatelmas) [reply format.go]: line ending is `\r\n` but you're appending `\n` later.

File: AUTHORS (hunk 1 above)
Comment (ferhatelmas) [reply AUTHORS]: ok, then

maybe another PR can remove it.

Comment (hyyking) [reply AUTHORS]: I removed it

File: Dockerfile (hunk 2 above)
Comment (ferhatelmas) [reply Dockerfile]: yes, instead having tests will be better.

This is dockerfile of the command and this comment looks suspicious.

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**